File: Keeper/Models/VaultKeep.cs
namespace Keeper.Models;

public class VaultKeep
{
  public int Id { get; set; }
  public int VaultId { get; set; }
  public int KeepId { get; set; }
  public string CreatorId { get; set; }
  public Keep Keep { get; set; }
  public Profile Creator { get; set; }
}

public class KeepInVault : Keep
{
  public int VaultKeepId { get; set; }
}

File: Keeper.client/src/models/VaultKeep.js
import { Profile } from "./Account.js"
import { Keep } from "./Keep.js"

export class VaultKeep {
  constructor(data) {
    this.id = data.id
    this.vaultId = data.vaultId
    this.keepId = data.keepId
    this.creatorId = data.creatorId
    this.keep = data.keep ? new Keep(data.keep) : null
    this.creator = data.creator ? new Profile(data.creator) : null
  }
}

export class KeepInVault extends Keep {
  constructor(data) {
    super(data)
    this.vaultKeepId = data.vaultKeepId
  }
}